auth/server/src/main/java/com/seagates3/authserver/AuthServerConfig.java

```diff
@@ -131,19 +133,41 @@ public static void logConfigProps() {
      * @throws GeneralSecurityException
      */
     public static void loadCredentials() throws GeneralSecurityException, Exception {
-
-        Properties authServerConfig = AuthServerConfig.authServerConfig;
-        String encryptedPasswd = authServerConfig.getProperty("ldapLoginPW");
-        Path keyStoreFilePath = getKeyStorePath();
-        PrivateKey privateKey = JKSUtil.getPrivateKeyFromJKS(
-                                keyStoreFilePath.toString(), getCertAlias(),
-                                getKeyStorePassword());
-        if (privateKey == null) {
-             throw new GeneralSecurityException("Failed to find Private Key ["
-                + keyStoreFilePath + "].");
-        }
-        ldapPasswd = RSAEncryptDecryptUtil.decrypt(encryptedPasswd,
-                     privateKey);
+       logger = LoggerFactory.getLogger(AuthServerConfig.class.getName());
+       logger.debug("Loading Openldap credentials");
+       String keystorePasswd = null;
+       String encryptedPasswd = authServerConfig.getProperty("ldapLoginPW");
+       Path keyStoreFilePath = getKeyStorePath();
+       try {
+         String cmd = authServerConfig.getProperty("cortexSecCmd");
+         Process cortxsec = Runtime.getRuntime().exec(cmd);
+         int exitVal = cortxsec.waitFor();
+         if (exitVal != 0) {
+           logger.debug("S3 Cipher util failed to return keystore password");
```

Comment:
This will not count as an error as this is an expected fallback mechanism. Internal mechanism of generating keystore password will not be highlighted. exitValue will not help as of now, but we can log it for debug purposes, no harm.

Reply:
got it.